common-http/src/main/java/io/cdap/common/http/HttpResponseConsumer.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.common.http;
+
+import java.nio.ByteBuffer;
+
+public abstract class HttpResponseConsumer {
+  // Default 64K chunk size
+  private static final int DEFAULT_CHUNK_SIZE = 65536;
+  private int chunkSize;
+
+  public HttpResponseConsumer() {
+    this.chunkSize = DEFAULT_CHUNK_SIZE;
+  }
+
+  public HttpResponseConsumer(int chunkSize) {
+    this.chunkSize = chunkSize;
+  }
+
+  /**
+   * This method is invoked when a new chunk of the response body is available to be consumed.
+   *
+   * @param chunk a {@link ByteBuffer} containing a chunk of the response body
+   */
+  public abstract void onReceived(ByteBuffer chunk);
```

Comment:
Should have a method to signal the end of the body transfer so that the implementation can perform appropriate action (e.g. closing the output file stream).

Reply:
added onFinished method.

Reply:
This method should returns a boolean to allow the consumer implementation choose to force closing the connection early without reading the entire content.